docs/USAGE-GUIDE.md

```diff
@@ -607,66 +607,29 @@ In TLS1.3, session ticket messages are sent after the handshake as "post-handsha
 
 Additionally, in TLS1.3, multiple session tickets may be issued for the same connection. Servers can call `s2n_config_set_initial_ticket_count()` to set the number of tickets they want to send and `s2n_connection_add_new_tickets_to_send()` to increase the number of tickets to send during a connection.
 
-### s2n\_config\_set\_client\_hello\_cb
+### Keying Material Lifetimes in Session Resumption
 
-```c
-int s2n_config_set_client_hello_cb(struct s2n_config *config, s2n_client_hello_fn client_hello_callback, void *ctx);
-```
-
-**s2n_config_set_client_hello_cb** allows the caller to set a callback function
-that will be called after ClientHello was parsed.
+Users may have questions on how long a secret generated from a full handshake is re-used with session resumption. The mechanism to expire old keying material varies based on the TLS version. In TLS1.2, the server only issues a new session ticket when doing a full handshake. Therefore the lifetime of the keying material inside the ticket is tied to the lifetime of the key used to encrypt the ticket. In TLS1.3, the encrypted ticket has an explicit lifetime stored in it, after which the keying material inside cannot be re-used. Servers can call `s2n_connection_set_server_keying_material_lifetime()` to configure this value.
 
-```c
-typedef int s2n_client_hello_fn(struct s2n_connection *conn, void *ctx);
-```
+## Client Hello Getters
 
-The callback function takes a s2n-tls connection as input, which receives the
-ClientHello and the context previously provided in **s2n_config_set_client_hello_cb**.
-The callback can access any ClientHello information from the connection and use
-the **s2n_connection_set_config** call to change the config of the connection.
+s2n-tls stores the received Client Hello and makes it available to users for logging purposes or to branch off of its content. Call `s2n_connection_get_client_hello()` to get a pointer the `s2n_client_hello` struct storing the Client Hello. A NULL value will be returned if the connection has not yet received the Client Hello. The earliest point in the handshake when this struct is available is during the [Client Hello Callback](#client-hello-callback). 
```

Comment:
```suggestion
s2n-tls stores the received Client Hello and makes it available to the application for logging and etc. Call `s2n_connection_get_client_hello()` to get a pointer the `s2n_client_hello` struct storing the Client Hello. A NULL value will be returned if the connection has not yet received the Client Hello. The earliest point in the handshake when this struct is available is during the [Client Hello Callback](#client-hello-callback). 
```

Reply:
```suggestion
s2n-tls stores the received Client Hello and makes it available to the application. Call `s2n_connection_get_client_hello()` to get a pointer to the `s2n_client_hello` struct storing the Client Hello message. A NULL value will be returned if the connection has not yet received the Client Hello. The earliest point in the handshake when this struct is available is during the [Client Hello Callback](#client-hello-callback). 
```
I think we clean the client hello up if they free the handshake-- might be worth mentioning.